### PR TITLE
move kubeadm api group testing to kubeadm package

### PR DESCRIPTION
**Pull request by mikedanese**
I think this is sufficient to at least preserve round trip testing.

### CODE REVIEW

**Comment by deads2k** [comment]
I think you've just duplicated what the existing `pkg/api` does, but I'm ok with the approach. If you created your own scheme, I think the separate would be better.  Something like https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/sample-apiserver/pkg/apiserver/apiserver.go#L43-L59 , but if you want to defer it, I can try to remember to mop it up later.

**Comment by mikedanese** [comment]
@deads2k does this look ok now?

File: cmd/kubeadm/app/apis/kubeadm/serialization_test.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+)
+
+const (
+	seed = 1
+)
+
+func TestRoundTripTypes(t *testing.T) {
+
+	groupFactoryRegistry := make(announced.APIGroupFactoryRegistry)
+	registry := registered.NewOrDie("kubeadmtest")
```
Comment (deads2k) [inline cmd/kubeadm/app/apis/kubeadm/serialization_test.go]: This isn't a name, its a list of version restritions.  Go ahead and leave it empty.

File: cmd/kubeadm/app/apis/kubeadm/serialization_test.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	scheme := runtime.NewScheme()
+
+	install.Install(groupFactoryRegistry, registry, scheme)
+	fuzzer := apitesting.FuzzerFor(kapitesting.FuzzerFuncs(t, api.Codecs), rand.NewSource(seed))
+	apitesting.RoundTripTypesWithoutProtobuf(t, scheme, api.Codecs, fuzzer, nil)
```
Comment (deads2k) [inline cmd/kubeadm/app/apis/kubeadm/serialization_test.go]: Add `Codecs               = serializer.NewCodecFactory(Scheme)` above and use it as the codec factory option

File: cmd/kubeadm/app/apis/kubeadm/serialization_test.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+)
+
+func TestRoundTripTypes(t *testing.T) {
+
+	groupFactoryRegistry := make(announced.APIGroupFactoryRegistry)
+	registry := registered.NewOrDie("kubeadmtest")
+	scheme := runtime.NewScheme()
+
+	install.Install(groupFactoryRegistry, registry, scheme)
+	fuzzer := apitesting.FuzzerFor(kapitesting.FuzzerFuncs(t, api.Codecs), rand.NewSource(seed))
```
Comment (deads2k) [inline cmd/kubeadm/app/apis/kubeadm/serialization_test.go]: This is ok for now, but you should add a TODO that once we've pulled kubeadm types of the main scheme, we should move the fuzzers funcs here.

**Comment by deads2k** [comment]
> @deads2k does this look ok now?

Minor comments, but that's what I was thinking of.

**Comment by mikedanese** [comment]
Round trips are failing with this cryptic error after moving off api.Codecs to a custom one.
```
        roundtrip.go:71: starting group "kubeadm.k8s.io"
        roundtrip.go:101: round tripping kubeadm.k8s.io/__internal, Kind=NodeConfiguration
        roundtrip.go:159:       round tripping to kubeadm.k8s.io/v1alpha1, Kind=NodeConfiguration v1alpha1.NodeConfiguration
        roundtrip.go:232: 3: NodeConfiguration: diff: 
                object.TypeMeta.Kind:
                  a: ""
                  b: "NodeConfiguration"
                object.TypeMeta.APIVersion:
                  a: ""
                  b: "kubeadm.k8s.io/v1alpha1"
                Codec: &versioning.codec{encoder:(*json.Serializer)(0xc4204bc040), decoder:(*recognizer.decoder)(0xc42050c020), convertor:runtime.unsafeObjectConvertor{Scheme:(*runtime.Scheme)(0xc4204bc000)}, creater:(*runtime.Scheme)(0xc4204bc000), copier:(*runtime.Scheme)(0xc4204bc000), typer:(*runtime.Scheme)(0xc42
04bc000), defaulter:(*runtime.Scheme)(0xc4204bc000), encodeVersion:schema.GroupVersions{schema.GroupVersion{Group:"kubeadm.k8s.io", Version:"v1alpha1"}}, decodeVersion:runtime.internalGroupVersioner{}}
```

**Comment by deads2k** [comment]
That's neat.  I'll debug with the sample-apiserver tomorrow morning.

**Comment by mikedanese** [comment]
@deads2k i suspect this might be because we are using the defaultConverter which isn't clearing those fields. Most other apigroups use generated conversions which clear those fields on the internal trip.

**Comment by dmmcquay** [comment]
@mikedanese @deads2k any luck figuring it out?

**Comment by deads2k** [comment]
> @mikedanese @deads2k any luck figuring it out?

I think so.  I opened https://github.com/mikedanese/kubernetes/pull/5

**Comment by deads2k** [comment]
lgtm

**Comment by luxas** [comment]
@mikedanese I think you can self-apply the lgtm and approved labels

**Review by dmmcquay — APPROVED** [review]

**Comment by luxas** [comment]
@deads2k and @dmmcquay approved this, and @mikedanese said that he was gonna add the labels, I'm doing it now in order to getting it merged during his night

**Comment by luxas** [comment]
This now unblocks #41104

**Pull request merged**